unlock-app/src/helpers/Locks.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * Given the address of a lock, returns its accompanying transaction
+ * @param transactionStore
+ * @param lockAddress
+ * @returns {object}
+ */
+export function getLockTransaction(transactionStore, lockAddress) {
+  if (transactionStore && transactionStore.all) {
+    for (let transaction in transactionStore.all) {
```

Comment:
This may come back and bite us if there are 10000 transactions... but by then we'll have other problems ;)

Reply:
Yeah, I imagine that we'll adapt this over time if that becomes a problem. (I'm pretty sure we'll need a cache and a supporting database by then, for one.)